ui/app/AppLayouts/Wallet/popups/AddAccountModal.qml

```diff
@@ -22,134 +22,142 @@ import "../panels"
-                }
-            }
-        }
+        readonly property int numOfItems: 100
+        readonly property int pageNumber: 1
 
-        property string derivedPathError: RootStore.derivedAddressesError
+        property string passwordValidationError: ""
+        property bool isPasswordCorrect: false
 
-        onDerivedPathErrorChanged: {
-            if(Utils.isInvalidPasswordMessage(derivedPathError))
-                root.passwordValidationError = qsTr("Wrong password")
+        function getDerivedAddressList() {
+            if(advancedSelection.expandableItem.addAccountType === SelectGeneratedAccount.AddAccountType.ImportSeedPhrase
+                    && !!advancedSelection.expandableItem.path
+                    && !!advancedSelection.expandableItem.mnemonicText) {
+                RootStore.getDerivedAddressListForMnemonic(advancedSelection.expandableItem.mnemonicText,
+                                                           advancedSelection.expandableItem.path, numOfItems, pageNumber)
+            } else if(!!advancedSelection.expandableItem.path && !!advancedSelection.expandableItem.derivedFromAddress
+                      && (passwordInput.text.length > 0)) {
+                RootStore.getDerivedAddressList(passwordInput.text, advancedSelection.expandableItem.derivedFromAddress,
+                                                advancedSelection.expandableItem.path, numOfItems, pageNumber)
+            }
         }
 
         function showPasswordError(errMessage) {
             if (errMessage) {
                 if (Utils.isInvalidPasswordMessage(errMessage)) {
-                    root.passwordValidationError = qsTr("Wrong password")
+                    d.passwordValidationError = qsTr("Wrong password")
+                    scroll.contentY = -scroll.padding
                 } else {
-                    accountError.text = errMessage;
-                    accountError.open();
+                    accountError.text = errMessage
+                    accountError.open()
                 }
             }
         }
 
-        property var waitTimer: Timer {
-            interval: 1000
-            running: false
-            onTriggered: {
-                d.getDerivedAddressList()
-            }
-        }
-
         function generateNewAccount() {
-            // TODO the loaidng doesn't work because the function freezes th view. Might need to use threads
+            // TODO the loading doesn't work because the function freezes the view. Might need to use threads
             nextButton.loading = true
-            if (!validate() || !advancedSelection.validate()) {
-                Global.playErrorSound();
+            if (!advancedSelection.validate()) {
+                Global.playErrorSound()
                 return nextButton.loading = false
```

Comment:
Shouldn't this be `return !nextButton.loading`?

Reply:
It's ok, `loading` is handled by `nextButton.enabled` property